torchaudio/datasets/tedlium.py

```diff
@@ -168,7 +168,8 @@ def _load_audio(self, path: str, start_time: float, end_time: float, sample_rate
         end_time = int(float(end_time) * sample_rate)
         if torchaudio.get_audio_backend() == "sox_io":
```

Comment:
note: 
```
backend = torchaudio.get_audio_backend()
if backend == "sox_io" or (backend == "soundfile" and not torchaudio.USE_SOUNDFILE_LEGACY_INTERFACE)
``` 

Will give a better performace on soundfile backend with the new interface.

Reply:
or more comprehensively
```
backend = torchaudio.get_audio_backend()
if backend == "soundfile" and torchaudio. USE_SOUNDFILE_LEGACY_INTERFACE:
    kwargs = {"offset": start_time, "num_frames": end_time - start_time}
else:
    kwargs = {"frame_offset": start_time, "num_frames": end_time - start_time}
return torchaudio.load(path, **kwargs)
```

Reply:
good point: we should make sure we have a test that catches whether datasets are ready for the new soundfile interface

Reply:
added the suggested change above, since i also added the soundfile test with the new backend.